modules/ting_reference/ting_reference.module

```diff
@@ -403,6 +416,29 @@ function ting_reference_field_formatter_view($entity_type, $entity, $field, $ins
   return $element;
 }
 
+/**
+ * Ajax callback to render single reference object.
+ *
+ * It's returned as JSON.
+ *
+ * @param int $entity_id
+ *   The id of the ting object entity to load.
+ * @param string $view_mode
+ *   The view mode that the ting object should be rendered.
+ */
+function _ting_reference_ajax($entity_id, $view_mode = 'teaser') {
+  // Load the ting objects found.
+  $entity = entity_load('ting_object', array($entity_id));
+  $ting_object = ting_object_view(reset($entity), $view_mode);
+
+  $build = array(
+    'id' => $entity_id,
+    'content' => render($ting_object),
+  );
+
+  drupal_json_output($build);
```

Comment:
This seems to be a very generic approach to rendering ting objects asynchronously. Could this and associated logic be moved to the `ting` module so it could be used more widely in the future?

Reply:
Moved into helper function.